Doc arq de software

HenriqueHida opened the pull request:
Commit da criação do documento de arquitetura de software


Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce to <repo_name>?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x ] I have read the CONTRIBUTING.md doc.
- [x ] I have added necessary documentation (if appropriate).
- [x ] Any dependent changes have been merged and published in downstream modules.

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

LuisMiranda10 commented on docs/ArquiteturaReutilizacao/arquiteturaDeSoftware.md:
```diff
@@ -0,0 +1,414 @@
+
+| Versão | Data       | Descrição                                                  | Autor(es)              | Revisor(es)   |
+| ------ | ---------- | ---------------------------------------------------------- | ---------------------- | ------------- |
+| `1.0`  | 12/08/2024 | Criação do documento e adição de todas as visões.                 | Henrique Hida           |        |
```
Aqui o histórico de versão deve seguir o padrão do projeto, logo, recomendo que coloque ao final do arquivo

LuisMiranda10 commented on docs/ArquiteturaReutilizacao/arquiteturaDeSoftware.md:
```diff
@@ -0,0 +1,414 @@
+
+1. **Documento de Arquitetura de Software.** Disponível em: <https://unbarqdsw2023-1.github.io/2023.1_G5_ProjetoRiHappy/#/4.arquiteturareutilizacao/padroes/padroesarquiteturais>. Acesso em: 12 ago. 2024. [1]
+
+2. **Spotfy Web API.**. Disponível em: <https://developer.spotify.com/documentation/web-api>. Acesso em: 14 ago. 2024. [2]
```
Mesma coisa da revisão sobre o histórico de versão. Para seguir o padrão do projeto, recomendo colocar ao final do arquivo, entretanto, antes do histórico de versão (só olhar outros documentos feitos pelos integrantes)

LuisMiranda10 commented on docs/ArquiteturaReutilizacao/arquiteturaDeSoftware.md:
```diff
@@ -0,0 +1,414 @@
+
+O Documento da Arquitetura de Software se aplica ao projeto do "My Music", trabalho da disciplina de Arquitetura e Desenho de Software, que consiste em uma aplicação de streaming de música.
+
+# 4. Definições, Acrônimos e Abreviações
+
+| Abreviação/Acrônimo                                           | Definição                                                                                                                                                                  |
+|-------------------------------------------------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| **MVC**                                                    | Model-View-Controller: Um padrão de projeto arquitetural que separa a aplicação em três componentes principais: Modelo, Visão e Controlador.                              |
+| **GRASP** - General Responsibility Assignment Software Patterns | Conjunto de padrões que oferecem diretrizes para atribuição de responsabilidades a classes e objetos em design orientado a objetos.                                      |
+| **GoF** - Gang of Four                                       | Grupo de quatro autores que publicaram o livro "Design Patterns: Elements of Reusable Object-Oriented Software", contendo 23 padrões de design essenciais para o desenvolvimento de software. |
+| **UML** - Unified Modeling Language                         | Linguagem de modelagem unificada utilizada para especificar, visualizar, construir e documentar artefatos de sistemas de software.                                        |
+| **DAS** - Diagrama de Arquitetura de Software               | Representação visual que demonstra como os componentes de um sistema de software estão organizados e como interagem entre si.                                          |
+| **DER** - Diagrama Entidade-Relacionamento                  | Diagrama utilizado para modelar a estrutura lógica de um banco de dados, mostrando as entidades e os relacionamentos entre elas.                                        |
```
Nossa arquitetura não tem nada sobre MVC. Inclusive recomendo que coloque sobre nosso tipo de estilo arquitetural utilizado no projeto (cliente-servidor).  

Além do mais, DAS não significa "Diagrama de Arquitetura de Software", e sim "Documento de Arquitetura de Software".

Para que o documento não fique muito extenso, utilize como parâmetro esse documento do projeto do semestre anterior (https://unbarqdsw2023-1.github.io/2023.1_G3_ProjetoMercadoLivre/#/ArquiteturaReutilizacao/DocumentoArquiteturaSoftware). Inclusive coloque ele na bibliografia por favor.

LuisMiranda10 committed:
Update arquiteturaDeSoftware.md

HenriqueHida committed:
Criação do documento de Arquitetura de Software

Commit da criação do documento de arquitetura de software

HenriqueHida committed:
hotfix imagens

HenriqueHida committed:
Update mkdocs e ArqSoftware. Add visao Integração

Update dos arquivos de Arquitetura de software e mkdocs e adição do documento de Visão de Integração

HenriqueHida committed:
Update bibliografia

LuisMiranda10 commented on docs/ArquiteturaReutilizacao/arquiteturaDeSoftware.md:
```diff
@@ -0,0 +1,101 @@
+
+# 3. Representação Arquitetural
+
+Para o back-end, a linguagem escolhida para o desenvolvimento foi Dart, utilizando o framework Flutter. A escolha foi motivada pela versatilidade do Flutter em permitir o desenvolvimento de uma aplicação multiplataforma com uma única base de código, além da familiaridade dos membros do grupo com a linguagem Dart. O uso do Flutter também permite uma integração suave entre a interface do usuário e  ...
+
+Como modelo arquitetural, foi adotado o padrão cliente-servidor. Nesse modelo, a aplicação é dividida em dois componentes principais: o cliente, que é responsável pela interface do usuário e interação com o usuário final; e o servidor, que lida com a lógica de negócios, processamento de dados e comunicação com o banco de dados. Essa separação facilita a escalabilidade e manutenção do sistema, per ...
```
Aqui você esqueceu de citar que estamos usando o firebase como nossa principal de dados e também para uma autenticação de usuário mais segura, utilizando métodos seguros, como por exemplo, o uso do algoritmo HASH dentro das senhas para maior segurança do usuário.  Faltou além disso, falar sobre a utilização da API do spotify, que é literalmente nosso ponto chave dentro do projeto.

HenriqueHida replied:
É verdade, documentei essa parte no documento de visão e acabei esquecendo de adicionar ao tópico de representação arquitetural. Vou atualizar o documento

LuisMiranda10 commented on docs/ArquiteturaReutilizacao/arquiteturaDeSoftware.md:
```diff
@@ -0,0 +1,101 @@
+| **GRASP** - General Responsibility Assignment Software Patterns | Conjunto de padrões que oferecem diretrizes para atribuição de responsabilidades a classes e objetos em design orientado a objetos.                                      |
+| **GoF** - Gang of Four                                       | Grupo de quatro autores que publicaram o livro "Design Patterns: Elements of Reusable Object-Oriented Software", contendo 23 padrões de design essenciais para o desenvolvimento de software. |
+| **UML** - Unified Modeling Language                         | Linguagem de modelagem unificada utilizada para especificar, visualizar, construir e documentar artefatos de sistemas de software.                                        |
+| **DAS** - Documento de Arquitetura de Software               | Representação visual que demonstra como os componentes de um sistema de software estão organizados e como interagem entre si.                                          |
```
A definição do DAS ainda está como definição de um diagrama de arquitetura de software. Sugiro olhar o site abaixo para que coloque a definição correta (inclusive coloque ele na bibliografia caso seja necessário).

https://awari.com.br/documento-de-arquitetura-de-software-exemplo-e-estrutura-para-a-documentacao-adequada/

HenriqueHida replied:
Resolvi os problemas apontados no commit [422f17d](https://github.com/UnBArqDsw2024-1/2024.1_G2_My_Music/pull/116/commits/422f17d5283fd4bf0a500c1d6c1f5806bdaa0321)

LuisMiranda10 replied:
Certo, seu PR vai ser aceito.

HenriqueHida committed:
Update Documento de Arquitetura de software
